apis/v1alpha2/gateway_types.go

```diff
@@ -141,6 +141,17 @@ type GatewaySpec struct {
 // combination of Hostname, Port, and Protocol. This will be enforced by a
 // validating webhook.
 type Listener struct {
+	// Name is the name of the Listener. If more than one Listener is present
+	// each Listener MUST specify a name. The names of Listeners MUST be unique
+	// within a Gateway.
+	//
+	// Support: Core
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	// +optional
+	Name *string `json:"name,omitempty"`
```

Comment:
I came very close to doing that, it would definitely be simpler to document, but it does make the simple case a bit more involved. Will bring this up at community meeting today.

Reply:
Forgot to bring this up at community meeting, but did ask in [Slack](https://kubernetes.slack.com/archives/CR0H13KGA/p1628622944049400). At this point I'm leaning towards copying what Service does here, though I did tweak this a bit so the field more closely resembles the port name field in upstream. Especially if/when we add this concept to Route rules, I think making it a requirement will be frustrating for the ~90% of use cases that won't need it.